gprimitives/src/lib.rs

```diff
@@ -0,0 +1,184 @@
+// This file is part of Gear.
+
+// Copyright (C) 2021-2024 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! Gear primitive types.
+
+#![no_std]
+#![warn(missing_docs)]
+#![doc(html_logo_url = "https://docs.gear.rs/logo.svg")]
+#![doc(html_favicon_url = "https://gear-tech.io/favicons/favicon.ico")]
+
+pub use gear_ss58::Ss58Address;
+
+pub mod macros;
+
+use core::{
+    fmt,
+    str::{self, FromStr},
+};
+use derive_more::{AsMut, AsRef, Display, From, Into};
+use gear_ss58::RawSs58Address;
+#[cfg(feature = "codec")]
+use {
+    primitive_types::H256,
+    scale_info::{
+        scale::{self, Decode, Encode, MaxEncodedLen},
+        TypeInfo,
+    },
+};
+
+/// The error type returned when conversion fails.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Display)]
+pub enum ConversionError {
```

Comment:
Too generic name. Consider
```suggestion
pub enum GPrimitiveError {
```